DOCSP-13806: add note about $uuid parsing in extended json (#62)
mongodb/docs-java: source/fundamentals/data-formats/document-data-format-extended-json.txt

.. _extended_json_example_section:

.. note::

>    The ``$uuid`` Extended JSON type is parsed from a string to a
>    :java-docs:`BsonBinary </apidocs/bson/org/bson/BsonBinary.html#<init>(java.util.UUID)>`
>    object of binary subtype 4. For a further discussion of ``$uuid`` field
>    parsing, see the
>    :spec:`special rules for parsing $uuid fields </extended-json.rst#special-rules-for-parsing-uuid-fields>`
>    section in the extended JSON specification.

Reviewer: ```suggestion
   The ``$uuid`` Extended JSON type is parsed from a String to a
   :java-docs:`BsonBinary </apidocs/bson/org/bson/BsonBinary.html#<init>(java.util.UUID)>`
   object of binary subtype 4. For more information about ``$uuid`` field
   parsing, see the
   :spec:`special rules for parsing $uuid fields </extended-json.rst#special-rules-for-parsing-uuid-fields>`
   section in the extended JSON specification.
```

Author: Thanks for the notes. I agree with both (in addition "for more information..." is used often elsewhere in the docs). Committing your suggestion. 

Going forward, I will try to stress simple sentences (and parallel structure), and appropriately capitalize language types.
